src/BatteryStateOfChargeService.cpp

```diff
@@ -43,21 +48,55 @@ BatteryStateOfChargeService::~BatteryStateOfChargeService()
 
 double BatteryStateOfChargeService::totalAmpHoursUsed() const
 {
-    return 0.0;
+    if (totalAH_>BATTERY_AMP_HOUR_CAPACITY)
+        return 0;
+    else if (totalAH_<0)
+        return BATTERY_AMP_HOUR_CAPACITY;
+    else
+        return BATTERY_AMP_HOUR_CAPACITY - totalAH_;
 }
 
 bool BatteryStateOfChargeService::isCharging() const
 {
-    return false;
+    return isCharging_;
 }
 
+
 QTime BatteryStateOfChargeService::timeWhenChargedOrDepleted() const
 {
-    return QTime::currentTime();
+    QTime null_time(0,0,0,0);
+    if (return_time.isNull())
+        return null_time;
```

Comment:
Instead of having a local variable for this, I think it would be just as clear and shorter to replace line 67 and 69 with:
`return QTime(0, 0);`